### PR TITLE
Add connection timeout

### PR DESCRIPTION
**Pull request by godchen0212**
Add connection timeout.
Issue #7202 

Signed-off-by: godchen <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/distributed/querynode/client/client.go [hunk 1]
```diff
@@ -64,7 +65,9 @@ func (c *Client) connect(retryOptions ...retry.Option) error {
 	connectGrpcFunc := func() error {
 		opts := trace.GetInterceptorOpts()
 		log.Debug("QueryNodeClient try connect ", zap.String("address", c.addr))
-		conn, err := grpc.DialContext(c.ctx, c.addr,
+		ctx, cancel := context.WithTimeout(c.ctx, 15*time.Second)
+		defer cancel()
```
Comment (xige-16) [inline internal/distributed/querynode/client/client.go:69]: Is connect immediately disconnected after creation?

Comment (godchen0212) [reply internal/distributed/querynode/client/client.go:69]: No, the cancel() controls the connection process and won't affect subsequent operations

**Pull request merged**